test/fetch_multi_test.rb

```diff
@@ -223,6 +223,49 @@ def test_fetch_multi_on_derived_model_raises
     end
   end
 
+  def test_fetch_multi_with_mixed_hits_and_misses_returns_only_readonly_records
+    IdentityCache.with_fetch_read_only_records do
+      cache_response = {}
+      cache_response[@bob_blob_key] = cache_response_for(@bob)
+      cache_response[@joe_blob_key] = nil
+      cache_response[@fred_blob_key] = cache_response_for(@fred)
+      fetch_multi = fetch_multi_stub(cache_response)
+      assert_equal [@bob, @joe, @fred], Item.fetch_multi(@bob.id, @joe.id, @fred.id)
+      assert fetch_multi.has_been_called_with?(@bob_blob_key, @joe_blob_key, @fred_blob_key)
+
+      Item.fetch_multi(@bob.id, @joe.id, @fred.id).each do |record|
+        assert record.readonly?, "Record #{record} wasn't readonly"
+      end
+    end
+  end
+
+  def test_fetch_multi_with_mixed_hits_and_misses_and_responses_in_the_wrong_order_returns_readonly
+    IdentityCache.with_fetch_read_only_records do
+      cache_response = {}
+      cache_response[@bob_blob_key] = nil
+      cache_response[@joe_blob_key] = nil
+      cache_response[@fred_blob_key] = cache_response_for(@fred)
+      fetch_multi = fetch_multi_stub(cache_response)
+      assert_equal [@bob, @joe, @fred], Item.fetch_multi(@bob.id, @joe.id, @fred.id)
+      assert fetch_multi.has_been_called_with?(@bob_blob_key, @joe_blob_key, @fred_blob_key)
+
+      Item.fetch_multi(@bob.id, @joe.id, @fred.id).each do |record|
+        assert record.readonly?, "Record #{record} wasn't readonly"
+      end
+    end
+  end
+
+  def test_fetch_multi_with_open_transactions_returns_non_readonly_records
+    IdentityCache.with_fetch_read_only_records do
+      Item.connection.expects(:open_transactions).at_least_once.returns(1)
```

Comment:
I'm not sure whether it's just historical, but why do we use this expectation here instead of an actual `Item.transaction` block? We're using the transaction block in similar tests, e.g. in `test/normalized_belongs_to_test.rb`.